packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayerContainer.js

```diff
@@ -16,24 +28,92 @@ function storeMapper (stores) {
 class InteractionLayerContainer extends Component {
   constructor () {
     super()
+
+    this.state = {
+      currentMark: null,
+      drawing: false,
+      markings: []
+    }
+
+    this.drawMark = this.drawMark.bind(this)
+    this.getEventOffset = this.getEventOffset.bind(this)
+    this.getNewMark = this.getNewMark.bind(this)
     this.onMouseDown = this.onMouseDown.bind(this)
     this.onMouseMove = this.onMouseMove.bind(this)
     this.onMouseUp = this.onMouseUp.bind(this)
   }
 
   componentDidMount () {
-    // TODO: We're simply logging the event stream here for now, but this will
-    // be passed to the active drawing tool for parsing
-    // const stream = this.props.drawing.eventStream
-    // stream.subscribe(z => console.log(z))
+    // const stream = this.props.eventStream
+    // stream.subscribe(z => console.log('z = ', z))
+    this.drawMark()
+  }
+
+  componentDidUpdate () {
+    this.drawMark()
+  }
+
+  drawMark () {
+    const { eventStream } = this.props
+
+    const stream = eventStream
+    stream.subscribe(event => {
+      const { currentMark, drawing, markings } = this.state
+      let newMark
+      // initial mark
+      if (markings.length === 0 && event.event === 'mousedown') {
+        newMark = this.getNewMark(event)
+        this.setState({ currentMark: newMark, drawing: true })
+      }
+      // move mark while drawing
+      if (drawing && event.event === 'mousemove') {
+        newMark = this.getNewMark(event)
+        this.setState({ currentMark: newMark })
+      }
+      if (drawing && event.event === 'mouseup') {
+        this.setState({ drawing: false })
+      }
+      // subsequent mark
+      if (!drawing && currentMark !== null && event.event === 'mousedown') {
+        const filteredMarkings = markings.filter(marking => marking.id !== currentMark.id)
+        filteredMarkings.push(currentMark)
+        newMark = this.getNewMark(event)
+        this.setState({ currentMark: newMark, drawing: true, markings: filteredMarkings })
+      }
+    })
+  }
+
+  getEventOffset (x, y) {
+    const { svg } = this.props
+
+    const svgEvent = svg.createSVGPoint()
+    svgEvent.x = x
+    svgEvent.y = y
+    const svgEventOffset = svgEvent.matrixTransform(svg.getScreenCTM().inverse())
+
+    return svgEventOffset
+  }
+
+  getNewMark (event) {
+    const { activeDrawingTool, activeStepTasks } = this.props
+
+    const [drawingTask] = activeStepTasks.filter(task => task.type === 'drawing')
+    const newMark = Object.assign({
+      id: Math.random(),
+      toolType: drawingTask.tools[activeDrawingTool].type
+    }, event)
+
+    return newMark
   }
 
   addToStream (event, type) {
-    this.props.drawing.addToStream({
+    const svgEventOffset = this.getEventOffset(event.clientX, event.clientY)
```

Comment:
Is it better to convert the event coords from client to svg before adding to the RxJs eventStream (current implementation) or when consuming?

Reply:
I'm not sure what would be best here. I'm going to do a bit of reading and get back to this question.

Reply:
After thinking about this, I think we want the stream to retain the original event in case we want to cancel it to prevent bubbling, so to that end, the conversion should happen when consuming.